vignettes/adtr.Rmd

```diff
@@ -0,0 +1,725 @@
+  highest_imputation = "D",
+  date_imputation = "first"
+) %>%
+  derive_vars_dy(
+    reference_date = RANDDT,
+    source_vars = exprs(ADT)
+  )
+```
+
+```{r, echo=FALSE}
+dataset_vignette(tr, display_vars = exprs(USUBJID, RANDDT, TRLNKID, TRDTC, ADT, ADTF, ADY))
+```  
+
+### Unscheduled visits
+
+If you have data collected at unscheduled visits, these could be identified as
+uniquely in your SDTM.  However, if this is not the case and your grouping lesions by visit then you will need to amend the below code to apply company specific algorithm to re-label unscheduled visits as per your specifications.
+
+```{r}
+tr <- mutate(
+  tr,
+  AVISIT = VISIT,
+  AVISITN = VISITNUM
+)
+```
+
+
+## Derive Parameters for Lesion Diameters (`LDIAMn` & `NLDIAMn`){#parameters}
+
+Depending on your company specifications you may either create parameters for Target Lesion diameter for lymph nodes (`LDIAMn`) and non-lymph nodes (`NLDIAMn`) in ADTR or similar dataset.
+
+The below examples show the set up of the parameter details and the analysis value & flag (`AVAL/ANL01FL`) variables.
+      
+```{r}
+adtr <- bind_rows(
+  tr %>%
+    filter(TRTESTCD == "LDIAM") %>%
+    mutate(
+      tmp_lesion_nr = str_sub(TRLNKID, 3),
+      PARAMCD = paste0("LDIAM", tmp_lesion_nr),
+      PARAM = paste("Target Lesion", tmp_lesion_nr, "Analysis Diameter")
+    ),
+  tr %>%
+    filter(TRTESTCD == "LPERP") %>%
+    mutate(
+      tmp_lesion_nr = str_sub(TRLNKID, 3),
+      PARAMCD = paste0("NLDIAM", tmp_lesion_nr),
+      PARAM = paste("Target Lesion", tmp_lesion_nr, "Analysis Perpendicular")
+    )
+) %>%
+  mutate(
+    PARCAT1 = "Target Lesion(s)",
+    PARCAT2 = "Investigator",
+    PARCAT3 = "Recist 1.1",
+    AVAL = TRSTRESN,
+    ANL01FL = if_else(!is.na(AVAL), "Y", NA_character_)
+  ) %>%
+  select(-tmp_lesion_nr)
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
+  arrange(adtr, USUBJID, TRLNKID, AVISITN),
+  display_vars = exprs(USUBJID, TRLNKID, AVISIT, PARAMCD, PARAM, AVAL, ANL01FL)
+)
+```
+
+## Derive Parameter for Sum of Diameter and Analysis Flag `ANL01FL` {#sdiam}
+
+Sum of Target Lesions Diameters is the sum of the diameters of both lymph and
+non-lymph nodes (sum of analysis values `AVAL` from source observations `LDIAMn` & `LNDIAMn`).   
+
+In this vignette, the sum of the diameters is calculated across all the
+available lesion measurements, and we use the analysis flag `ANL01FL` to select
+the records that should be used as part of the analysis based on RECIST 1.1.
+Subsequently we then build on this analysis flag (see [`ANLxxFL`]) (#analflag) to
+demonstrate additional analysis flags for example purposes only. The analysis
+flags therefore should be setup according to your company specifications.
+
+
+In the example below, the lesion sizes are counted and grouped according to the by variables specified, in this case we group by the visit (`AVISIT` and `AVISIT`) to ensure that the correct lesion measurements are summed together. 
```

Comment:
AVISIT is coming 2 times, I think you mean to say AVISITN?

Reply:
Fixed